src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let res = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let word = (this[i] === undefined || this[i] === null) ? '' : this[i];
+
+      if (i === this.length - 1) {
+        res += word;
+      } else {
+        res += word += String(separator);
```

Comment:
Don't repeat yourself: move res += word out of this if / else and add separator after all elements except the last